libcnb-test/src/lib.rs

```diff
@@ -13,373 +13,131 @@ mod container_port_mapping;
-                }
-                BuildpackReference::Other(id) => pack_command.buildpack(id.clone()),
-            };
-        }
-
-        let output = Command::from(pack_command)
-            .stdout(Stdio::piped())
-            .stderr(Stdio::piped())
-            .spawn()
-            .expect("Could not spawn external 'pack' process")
-            .wait_with_output()
-            .expect("Error while waiting on external 'pack' process");
-
-        let integration_test_context = IntegrationTestContext {
-            pack_stdout: String::from_utf8_lossy(&output.stdout).into_owned(),
-            pack_stderr: String::from_utf8_lossy(&output.stderr).into_owned(),
-            image_name,
-            app_dir: PathBuf::from(temp_app_dir.path()),
-            integration_test: self,
-        };
-
-        if output.status.success() {
-            f(integration_test_context);
-        } else {
-            panic!(
-                "pack command failed with exit-code {}!\n\npack stdout:\n{}\n\npack stderr:\n{}",
-                output
-                    .status
-                    .code()
-                    .map_or(String::from("<unknown>"), |exit_code| exit_code.to_string()),
-                integration_test_context.pack_stdout,
-                integration_test_context.pack_stderr
-            )
-        }
+    pub fn run_test<C: Borrow<TestConfig>, F: FnOnce(TestContext)>(self, config: C, f: F) {
+        self.runner
+            .run_test_internal(self.image_name.clone(), config, f);
     }
-}
-
-/// Context for a currently executing integration test.
-pub struct IntegrationTestContext<'a> {
-    /// Standard output of `pack`, interpreted as an UTF-8 string.
-    pub pack_stdout: String,
-    /// Standard error of `pack`, interpreted as an UTF-8 string.
-    pub pack_stderr: String,
-    /// The name of the image this integration test created.
-    pub image_name: String,
-    /// The directory of the app this integration test uses.
-    ///
-    /// This is a copy of the `app_dir` directory passed to [`IntegrationTest::new`] and unique to
-    /// this integration test run. It is safe to modify the directory contents inside the test.
-    pub app_dir: PathBuf,
 
-    integration_test: &'a IntegrationTest,
-}
-
-impl<'a> IntegrationTestContext<'a> {
-    /// Prepares a new container with the image from the integration test.
+    /// Starts a subsequent integration test run with inherited configuration.
     ///
-    /// This will not create nor run the container immediately. Use the returned
-    /// `PrepareContainerContext` to configure the container, then call
-    /// [`start_with_default_process`](PrepareContainerContext::start_with_default_process) on it
-    /// to actually create and start the container.
+    /// This function is the same as [`TestContext::run_test`] but automatically inherits the
+    /// configuration from the previous test run. See [`TestContext::run_test`] for details.
     ///
```

Comment:
Should `run_test_inherit_config()` have an example in its rustdoc too? 

I had to copy-paste from the PR description to construct an example in one of the other review comments, since I couldn't find one in the codebase :-)

Reply:
As discussed in https://github.com/heroku/libcnb.rs/pull/422#discussion_r905110888, the function will be removed altogether.